### PR TITLE
speed up hardswish and hardsigmoid tests

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38256 speed up hardswish and hardsigmoid tests**

Summary:

Removes hypothesis to speed these tests up, as these tests were flagged as top slow
tests in CI. At the same time, combines the fbgemm and qnnpack test
cases together for better reuse.

Test Plan:

```
python test/test_quantization.py TestQuantizedOps.test_hardswish
python test/test_quantization.py TestQuantizedOps.test_qhardsigmoid
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21506831](https://our.internmc.facebook.com/intern/diff/D21506831)

### CODE REVIEW

File: test/quantization/test_quantized_op.py [hunk 1]
```diff
@@ -320,12 +267,46 @@ def test_qsigmoid(self, X):
...
-    @given(X=hu.tensor(shapes=hu.array_shapes(1, 5, 1, 5),
-                       elements=hu.floats(-1e3, 1e3, allow_nan=False, allow_infinity=False),
-                       qparams=hu.qparams()))
-    def test_qhardsigmoid(self, X):
-        _test_hardsigmoid(self, X, 'fbgemm')
+    def test_qhardsigmoid(self):
+        max_sides = (3, 5)
+        side_lens = (1, 7, 8)
+        torch_types = (torch.quint8, torch.qint8)
+        engines = ('fbgemm', 'qnnpack')
```
Comment (supriyar) [inline test/quantization/test_quantized_op.py:274]: Maybe use `@override_qengines` instead?

File: test/quantization/test_quantized_op.py [hunk 2]
```diff
@@ -320,12 +267,46 @@ def test_qsigmoid(self, X):
...
+                output_scale = (f_max - f_min) / (q_max - q_min + 1.0)
+                output_zero_point = 0 if torch_type == torch.qint32 else q_min
+                dqY_hat = F.hardsigmoid(dqX)
+                qY_hat = torch.quantize_per_tensor(dqY_hat, scale=output_scale,
+                                                   zero_point=output_zero_point,
+                                                   dtype=torch_type)
+
+                qY = torch.nn.quantized.functional.hardsigmoid(qX)
+                self.assertEqual(qY, qY_hat,
+                                 message="Hardsigmoid failed: {} vs. {}".format(qY, qY_hat))
```
Comment (supriyar) [inline test/quantization/test_quantized_op.py:309]: Might help for debugging if we specify which backend this failed on.

**Comment by malfet** [comment]
Change looks good to me, but outside of CI environment test already finish in a reasonable amount of time:
```
$ python test/test_quantization.py TestQuantizedOps.test_hardswish TestQuantizedOps.test_qhardsigmoid -v
test_hardswish (quantization.test_quantized_op.TestQuantizedOps) ... /home/nshulga/git/pytorch/test/quantization/test_quantized_op.py:101: UserWarning: 0quantize_tensor_per_tensor_affine current rounding mode is not set to round-to-nearest-ties-to-even (FE_TONEAREST). This will cause accuracy issues in quantized models. (Triggered internally at  /pytorch/aten/src/ATen/native/quantized/affine_quantizer.cpp:25.)
  dtype=torch_type)
ok
test_qhardsigmoid (quantization.test_quantized_op.TestQuantizedOps) ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.619s

OK
```

**Comment by malfet** [comment]
I've tried selecting different profiles, as defined in https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_utils.py#L659 but still can not get to 120 sec times reported during CircleCI runs

**Review by malfet — APPROVED** [review]

File: test/quantization/test_quantized_op.py [hunk 3]
```diff
@@ -320,12 +267,44 @@ def test_qsigmoid(self, X):
...
+        torch_types = (torch.quint8, torch.qint8)
+        combined = [max_sides, side_lens, torch_types]
+        test_cases = itertools.product(*combined)
+        for test_case in test_cases:
+            max_side, side_len, torch_type = test_case
+
+            if torch.backends.quantized.engine == 'qnnpack' and torch_type != torch.quint8:
+                continue
+
+            with override_quantized_engine(engine):
```
Comment (supriyar) [inline test/quantization/test_quantized_op.py:283]: This can be removed as well. The override will take care of it at the test level.

**Comment by malfet** [comment]
I know why tests can be slow when run by CI, but fast during single runs:
`hu.array_shapes(1, 8, 1, 8)` returns randomized tensor shape which can be as large as 16 million elements (i.e. `8**8`)
During individual test runs those values are always the same(since we fix the seed), and by chance quite small.
But since the seed is not reset between the runs it can generate a very different (and apparently quite large) tensor shapes when run in sequence with other tests.

**Comment by malfet** [comment]
@vkuzo you might need to rebase your change against latest master as I've landed a PR that adds limit to max number of element for hypothesis generated tensors (and it touches the same files are your diff)

**Comment by vkuzo** [comment]
> I know why tests can be slow when run by CI, but fast during single runs:
> `hu.array_shapes(1, 8, 1, 8)` returns randomized tensor shape which can be as large as 16 million elements (i.e. `8**8`)
> During individual test runs those values are always the same(since we fix the seed), and by chance quite small.
> But since the seed is not reset between the runs it can generate a very different (and apparently quite large) tensor shapes when run in sequence with other tests.

thanks for the find, makes sense!
